setup.sh

```diff
@@ -1,133 +1,135 @@
-if [[ $MIN_OS_VERSION != "$(printf "$MIN_OS_VERSION\n$CURRENT_OS_VERSION" | sort -V | sed -n 1p)" ]]; then
-    red_echo "MacOS minimum required version is ${MIN_OS_VERSION}. The installed version is ${CURRENT_OS_VERSION}. Please update your OS before running this script."
+if [[ "$kernel_name" != 'Darwin' ]] && [[ "$kernel_name" != 'Linux' ]]; then
+    red_echo "This script is only supported on MacOS and Linux."
     exit
 fi
 
-if [[ -z "${INTEGRATION_PORTAL_URL}" ]]; then
+if [[ "$kernel_name" == 'Darwin' ]]; then
+  CURRENT_OS_VERSION=$(sw_vers -productVersion)
+  # use version sorting to check if the current version is less than $MIN_OS_VERSION
+  if [[ $MIN_OS_VERSION != "$(printf "$MIN_OS_VERSION\n$CURRENT_OS_VERSION" | sort -V | sed -n 1p)" ]]; then
+      red_echo "MacOS minimum required version is ${MIN_OS_VERSION}. The installed version is ${CURRENT_OS_VERSION}. Please update your OS before running this script."
+      exit
+  fi
+fi
+
+if [[ "${CONFIGURE_LEAPP}" != "0" ]] && [[ -z "${INTEGRATION_PORTAL_URL}" ]]; then
     red_echo "INTEGRATION_PORTAL_URL must be provided"
     exit
 fi
 
-if [[ -z "${LEAPP_ROLES}" ]]; then
+if [[ "${CONFIGURE_LEAPP}" != "0" ]] && [[ -z "${LEAPP_ROLES}" ]]; then
     red_echo "LEAPP_ROLES must be provided"
     exit
 fi
 
 # Install Homebrew if not installed
 # This may optionally install the Xcode CLT if it is not already installed.
-which -s brew
-if [[ $? != 0 ]] ; then
-    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+if [[ "$kernel_name" == 'Darwin' ]] && ! which brew > /dev/null ; then
+    # If using Linux, create /home/<user>/ using sudo permission
+    if [[ "$kernel_name" == "Linux" ]] && [[ ! -e "/home/$(whoami)" ]]; then
+        sudo mkdir -p "/home/$(whoami)"
+        if id -gn | grep 'users' > /dev/null; then
+            group='users'
+        else
+            group=$(id -gn | cut -d ' ' -f 1)
+        fi
+        sudo chown -R "$(whoami):$group" "/home/$(whoami)"
+    fi
+
+    NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
 
     # If using an M1 machine, load shell environment to run brew commands
-    if [[ $(uname -m) == 'arm64' ]]; then
-        echo ‘# Set PATH, MANPATH, etc., for Homebrew.’ >> ~/.zprofile
-        echo ‘eval "$(/opt/homebrew/bin/brew shellenv)"’ >> ~/.zprofile
+    if [[ "$kernel_name" == "Darwin" ]] && [[ $(uname -m) == 'arm64' ]]; then
+        echo '# Set PATH, MANPATH, etc., for Homebrew.' >> ~/.zprofile
+        echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' >> ~/.zprofile
         eval "$(/opt/homebrew/bin/brew shellenv)"
     fi
+
+    if [[ "$kernel_name" == "Linux" ]]; then
+        (echo; echo 'eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"') >> ~/.bashrc
+        eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+        sudo apt install -y build-essential
+        brew install gcc
```

Comment:
If we can install gcc without brew, we may not need `linuxbrew` at all.  Is there a workaround?

Reply:
We could. Actually this code is unreachable and we could just remove it outright too -- the Brew installation if check above first checks if we're on Mac OS, so if we're on Linux Brew doesn't get installed and this code doesn't run

Reply:
Do we need Linuxbrew for other purposes?

Reply:
I haven't run anything setting up the eng image that required Linuxbrew.